main.go

```diff
@@ -173,15 +194,18 @@ func fetchMetricFamilies(url string, ch chan<- *dto.MetricFamily) {
 }
 
 func main() {
-	runtime.GOMAXPROCS(2)
-	if len(os.Args) != 2 {
-		log.Fatalf("Usage: %s METRICS_URL", os.Args[0])
-	}
-
+	runtime.GOMAXPROCS(4)
+	cert := flag.String("cert", "", "certificate file")
+	key := flag.String("key", "", "key file")
+	flag.Parse()
 	mfChan := make(chan *dto.MetricFamily, 1024)
-
-	go fetchMetricFamilies(os.Args[1], mfChan)
-
+	if flag.NFlag() == 0  && (len(flag.Args()) == 1 && flag.Args()[0] != "") {
+		go fetchMetricFamilies(flag.Args()[0], mfChan)
+	} else if flag.NFlag() == 2 && (*cert != "" && *key != "") {
+		go fetchMetricFamiliesWithAuth(flag.Args()[0], mfChan, *cert, *key)
+	} else {
+		log.Fatalf("Usage: %s METRICS_URL\n with authentication: %s -cert=/path/to/certificate -key=/path/to/key METRICS_URL", os.Args[0], os.Args[0])
+	}
```

Comment:
Lines 202 to 208 are a bit clunky. Also, the check will not work that way anymore should we ever add more flags.

Suggestion is to do the check as
```Go
if (certificate != "" && key == "") || (certificate == "" && key != "") {
	log.Fatalf("as above")
}
```

Then have only one function `fetchMetricFamilies(url string, ch chan<- *dto.MetricFamily, certificate string, key string)` being called here as `go fetchMetricFamilies(flag.Args()[0], mfChan, *cert, *key)`.

In that function, do something like:

```Go
func fetchMetricFamilies(url string, ch chan<- *dto.MetricFamily, certificate string, key string) {
	defer close(ch)

	var transport *http.Transport
	if certificate != "" && key != "" {
		cert, err := tls.LoadX509KeyPair(certificate, key)
		if err != nil {
			log.Fatal(err)
		}
		tlsConfig := &tls.Config{
			Certificates: []tls.Certificate{cert},
		}
		tlsConfig.BuildNameToCertificate()
		transport := &http.Transport{TLSClientConfig: tlsConfig}
	}
	client := &http.Client{Transport: transport}
	decodeContent(client, url, ch)
```